src/lsstseries/analysis/structure_function_calculators/structure_function_calculator.py

```diff
@@ -0,0 +1,93 @@
+from typing import List
+
+import numpy as np
+
+from lsstseries.analysis.structure_function_argument_containers.structure_function_argument_container import (
+    StructureFunctionArgumentContainer,
+)
+
+
+class StructureFunctionCalculator:
```

Comment:
If we make this a subclass of python's `AbstractBaseClass` then we could use the `@abstractmethod` decorator on the `calculate` method. And I _think_ on the `name_id` and `expected_argument_container` static methods. Doing so would require subclasses of `StructureFunctionCalculator` to implement those methods. If the subclasses failed to implement those methods, then the interpreter would prevent new instances of those classes from being created.

As it stands now, we only raise `NotImplementedError`s which are nice, but don't really enforce anything.

Reply:
+1 to abstract methods.

Reply:
Added ABC and @abstractmethod annotations.